faiss/impl/HNSW.cpp

```diff
@@ -660,6 +662,11 @@ int search_from_candidates(
         }
     }
 
+    // Completed collection of result. Run post processor.
+    collector->finalize(nres, I);
```

Comment:
should we rename this function to postprocess then?

Reply:
I used `finalize` to express the intention that it will be called only once and cannot collect more result after it is called but I can change the method name.

Reply:
> it will be called only once

There is no guarantee around this. We should add documentation around this also, I think better name is still post_process.